index.js

```diff
@@ -78,7 +77,15 @@ class PreloadPlugin {
         }).forEach(entry => {
           entry = `${publicPath}${entry}`;
           if (options.rel === 'preload') {
-            filesToInclude+= `<link rel="${options.rel}" href="${entry}" as="${options.as}">\n`;
+            // If `as` value is not provided in option, dynamically determine the correct
+            // value depends on suffix of filename. Otherwise use the given `as` value.
+            let asValue;
+            if (!options.as) {
+              asValue = entry.match(/\.css$/) ? 'style' : 'script';
```

Comment:
If we're going to do extension based `as` type inference, I wonder if we should extend this to also include other types from the spec: https://w3c.github.io/preload/#link-element-extensions (fonts, images, video). Thoughts?